validation/src/main/scala/hmda/validation/dsl/PredicateGeo.scala

```diff
@@ -0,0 +1,114 @@
+package hmda.validation.dsl
+
+import hmda.model.census.{ CBSAMetroMicroLookup, CBSATractLookup }
+import hmda.model.fi.lar.Geography
+
+object PredicateGeo {
+  val cbsaTracts = CBSATractLookup.values
+
+  val cbsaMetroMicro = CBSAMetroMicroLookup.values
+
+  val validMsaCombinationSet = cbsaTracts.map { cbsa =>
+    (cbsa.geoIdMsa, cbsa.state, cbsa.county, cbsa.tractDecimal)
+  }.toSet
+
+  val validMdCombinationSet = cbsaTracts.map { cbsa =>
+    (cbsa.metDivFp, cbsa.state, cbsa.county, cbsa.tractDecimal)
+  }.toSet
+
+  val validMsaCombinationSetNoTract = cbsaTracts.map { cbsa =>
+    (cbsa.geoIdMsa, cbsa.state, cbsa.county)
+  }.toSet
+
+  val validMdCombinationSetNoTract = cbsaTracts.map { cbsa =>
+    (cbsa.metDivFp, cbsa.state, cbsa.county)
+  }.toSet
+
+  val hasMsaSet = cbsaTracts.map { cbsa =>
+    (cbsa.geoIdMsa, cbsa.state, cbsa.county)
+  }.filter(x => x._1 != "").map { x =>
+    (x._2, x._3)
+  }.toSet
+
+  val hasMdSet = cbsaTracts.map { cbsa =>
+    (cbsa.metDivFp, cbsa.state, cbsa.county)
+  }.filter(x => x._1 != "").map { x =>
+    (x._2, x._3)
+  }.toSet
+
+  val MsaNotMicro = cbsaMetroMicro.map { cbsa =>
+    (cbsa.GEOIOD, cbsa.MEMI)
+  }.filter(x => x._2 == 1).map { x =>
+    x._1
+  }
+
+  val hasMsaNotMicroSet = cbsaTracts.map { cbsa =>
+    (cbsa.geoIdMsa, cbsa.state, cbsa.county)
+  }.filter(x => MsaNotMicro.contains(x._1)).map { x =>
+    (x._2, x._3)
+  }.toSet
```

Comment:
I propose something like:
```scala
  private val msaNotMicro = cbsaMetroMicro.filter(x => x.MEMI == 1).map { x => x.GEOIOD }.toSet
  val hasMsaNotMicroSet = cbsaTracts.filter(x => msaNotMicro.contains(x.geoIdMsa)).map { x =>
    (x.state, x.county)
  }.toSet
```
Maybe the second can also be private, and maybe you'd want to use underscore instead of `x` in some of those places, but that's the basic idea: do the mapping and then the tuple-ing so the names are still meaningful, and make sure the first one is a Set.